pkg/parser/threefour/parser.go

```diff
@@ -231,7 +231,7 @@ func (p *Parser) resolveEndpoint(ip net.IP, securityIdentity uint64) *pb.Endpoin
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			return &pb.Endpoint{
 				ID:        ep.GetID(),
-				Identity:  securityIdentity,
+				Identity:  uint64(ep.GetIdentity()),
```

Comment:
Why is `Endpoint.Identity` defined as a `uint64` and not a `uint32` like in `identity.NumericIdentity` ? At first glance, it seems that these type casts could be avoided.

Reply:
Good question. Unfortunately, neither Cilium nor Hubble are very consistent here (the `Endpoint.ID` is also a `uint64` in the interfaces and protobuf definitions, but internally and in the datapath a 16bit unsigned).

What we can do is change the data type of pb.Endpoint to `uint32`. It would be a breaking protobuf change though, something we should do before a proper release.

Reply:
Ah, thinking about it, the reason is probably that the auto-generated OpenAPI definitions from Cilium use `int64` for any `integer` types, so we adapted this without checking the actual data size. We should revisit the integer types we use in the protobuf and make sure they are appropriately sized.

Reply:
Created https://github.com/cilium/hubble/issues/158